packages/components/Button/src/Button.tsx

```diff
@@ -69,18 +72,37 @@ export const Button = compose<ButtonType>({
           }
         });
       }
-      const label = accessibilityLabel ?? childText;
 
-      return (
-        <Slots.root {...mergedProps} accessibilityLabel={label}>
+      const label = accessibilityLabel ?? childText;
+      const buttonContent = (
+        <React.Fragment>
           {loading && <ActivityIndicator />}
           {shouldShowIcon && iconPosition === 'before' && <Slots.icon {...iconProps} />}
           {React.Children.map(children, (child) =>
             typeof child === 'string' ? <Slots.content key="content">{child}</Slots.content> : child,
           )}
           {shouldShowIcon && iconPosition === 'after' && <Slots.icon {...iconProps} />}
-        </Slots.root>
+        </React.Fragment>
       );
+
+      const hasRipple = Platform.OS === 'android';
+      if (hasRipple) {
+        const [outerStyleProps, innerStyleProps] = extractOuterStylePropsAndroid(mergedProps.style);
+        return (
```

Comment:
I still feel like this whole "hasRipple" logic can be extracted to a simple function component / stagedComponent so that it can be re-used across multiple components, rather than copied for every new component that has a ripple. I'm sure that will be most controls.

Reply:
Based on offline discussions, would like to continue with the current approach for now. When the requirement for the ripple on other components arises, will pick up work on completing the ripple as a component (currently in [draft](https://github.com/microsoft/fluentui-react-native/pull/2273)